mlabs/nft-endpoint-spec.md

```diff
@@ -0,0 +1,51 @@
+# NFT Contract Specification
+
+This project adapts the Ethereum-style approach to NFTs as a digital certificate of authenticity or ownership, it allows a creator to make a digital asset representing some artwork, then sell the asset, and for owners of the asset to be confirmed.
+
+ownership can only be transferred through the contract, so when the asset is re-sold, a royalty to the artist can be enforced
+
+## Author Contact
+
+### StartParams
+
+prerequisite: none
+
+input:
+Mlabs.Nft.Contract.Api.StartParams
+(content, share, price)
+
+behavior:
+instantiates the 'User' Contract, which represents an asset described by `content`
+the author is set to the original owner 
+the entire `StartParams` will need to be kept as some internal state to be referenced/updated, along with the author and the current owner
+
+## User Contract
+
+all endpoints on this contract presume that AuthorContract.StartParams has been called. 
+
+### SetPrice
+
+prerequiste: none beyond contract instantiation
+must be the current owner
+
+input:
+Mlabs.Nft.Contract.Api.SetPrice
+
+behavior:  
+updates the `price` parameter needed for the `Buy` endpoint
```

Comment:
Maybe it's worth to say that we can lock the NFT by setting Nothing as a price

Reply:
sure.